[Enhancement] Add bonus to views and some renamings

fde312 opened the pull request:
<!-- Thanks for contributing to Artemis! Before you submit your pull request, please make sure to check the following boxes by putting an x in the [ ] (don't: [x ], [ x], do: [x]) -->
<!-- If your pull request is not ready for review yet, create a draft pull request! -->

### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) locally.
- [x] Client: I added multiple screenshots/screencasts of my UI changes
- [x] Client: I translated all the newly inserted strings into German and English

### Motivation and Context
<!-- Why is this change required? What problem does it solve? -->
<!-- If it fixes an open issue, please link to the issue here. -->
As we added bonus points to exercises, we want to display them as well in different views.

### Description
<!-- Describe your changes in detail -->
- Added bonus points to some views
- Changed some translations (Exercise Type -> Type, Max Points -> Points)

### Steps for Testing
<!-- Please describe in detail how the reviewer can test your changes. -->

1. Log in to Artemis
2. Navigate to Course Administration and 
3. Go to Assessment/Instructor Dashboard of a course and then to Exercise Dashboard 
4. Check if for exercises with bonus points the bonus points are displayed as well also (Exercise Type -> Type, Max Points -> Points)
5. Navigate to Course Overview and go into a course
6. Enter a Exercise and check if for exercises with bonus points the bonus points are displayed as well
7. Go to Course Administration and then to Exams
8. Check Exercise Groups and check that it says Points and not Max. Points
9. Participate in Exam (or do a test run) check if for exercises (modeling, text and programming) with bonus points the bonus points are displayed as well

Check also if I missed some other spots where this can be changed/added (see also comments in discussion). Just let me know :) 
### Test Coverage
<!-- Please add the test coverage for all changes files here. You can see this when executing the tests locally (see build.gradle and package.json) or when looking into the corresponding Bamboo build plan -->
Nothing changed

### Screenshots
<!-- Add screenshots to demonstrate the changes in the UI. -->
<!-- Create a GIF file from a screen recording in a docker container https://toub.es/2017/09/11/high-quality-gif-with-ffmpeg-and-docker/ -->

Student Exercise Details View:
Bonus:
<img width="1680" alt="image" src="https://user-images.githubusercontent.com/41108487/97758822-02736900-1b00-11eb-817a-3b0c8572b6a5.png">

No Bonus:
<img width="1676" alt="image" src="https://user-images.githubusercontent.com/41108487/97758861-174ffc80-1b00-11eb-80a9-ab62fd0c3091.png">

Assessment Dashboard:
<img width="1675" alt="image" src="https://user-images.githubusercontent.com/41108487/97758899-29319f80-1b00-11eb-9fc3-a2987d9123fc.png">

Instructor Dashboard:
<img width="1679" alt="image" src="https://user-images.githubusercontent.com/41108487/97758926-35b5f800-1b00-11eb-91df-0f8f2cd3388e.png">

Exam:
<img width="1399" alt="image" src="https://user-images.githubusercontent.com/41108487/97758990-60a04c00-1b00-11eb-8d24-5e467dc092be.png">

<img width="1654" alt="image" src="https://user-images.githubusercontent.com/41108487/97759048-7f9ede00-1b00-11eb-9fa1-c09e935a191b.png">

fde312 commented:
Client tests are failing because of formatting. I just runned prettier and got this back:
<img width="893" alt="image" src="https://user-images.githubusercontent.com/41108487/97760253-005ed980-1b03-11eb-81df-792a23a21f61.png">

So do not wonder why it is formatted so strangely 😅

MaisiKoleni reviewed (commented):
Looks good and works well (for both German and English).

I just have some points that are unclear to me:
- Do we want to display bonus points in the "View" exercise as well? Currently, there are only the normal points.
- For exams (in this case a test run) i found "Max. Points" in the summary. Is this intended/ok?
  ![image](https://user-images.githubusercontent.com/11130248/97803801-090ef700-1c4c-11eb-9735-bd3e9147dbbb.png)

FrankeLukas commented on src/main/webapp/app/exam/participate/exercises/modeling/modeling-exam-submission.component.html:
```diff
@@ -1,7 +1,11 @@
     <!--region Left Panel-->
     <span left-header *ngIf="exercise.title">{{ exercise.title }}</span>
-    <span left-header>&nbsp;[{{ exercise.maxScore }} {{ 'artemisApp.examParticipation.points' | translate }}]</span>
+    <span left-header
+        >&nbsp;[{{ exercise.maxScore }} {{ 'artemisApp.examParticipation.points' | translate }}
```
Can we move the closing tags onto the same line? (Here and in the files below.) Imho
```
<span
    >
```
looks very weird compared to just having `<span>` when there are no attributes.

FrankeLukas reviewed (commented):
Imho
![grafik](https://user-images.githubusercontent.com/72132281/97850606-8ee67d00-1cf4-11eb-9d18-717683d4ac54.png)
would look better as
![grafik](https://user-images.githubusercontent.com/72132281/97850813-d3721880-1cf4-11eb-91de-12d28a0610ad.png)
Opinions?

krusche commented:
> Imho
> ![grafik](https://user-images.githubusercontent.com/72132281/97850606-8ee67d00-1cf4-11eb-9d18-717683d4ac54.png)
> would look better as
> ![grafik](https://user-images.githubusercontent.com/72132281/97850813-d3721880-1cf4-11eb-91de-12d28a0610ad.png)
> Opinions?

I would like to avoid a third line and rather make the header a bit smaller. I think the current display is good

krusche commented:
> * Do we want to display bonus points in the "View" exercise as well? Currently, there are only the normal points.
Yes, this would be good!
> * For exams (in this case a test run) i found "Max. Points" in the summary. Is this intended/ok?
Please rename to Points here as well.

Basically Max. Points should not be used any more in the whole application

FuchsDominik reviewed (changes requested):
In my exam summary, the bonus points are not shown. (Exercise 1 should have bonus points)
![image](https://user-images.githubusercontent.com/44401560/97865755-fb6c7680-1d0a-11eb-823a-83a7d59200e9.png)
Mini remark: You could switch the white space to the second string, so without the second string, the square brackets are even:
![image](https://user-images.githubusercontent.com/44401560/97866103-8c435200-1d0b-11eb-8633-5ecc6034cb0a.png)
The rest looked very good to me!

fde312 commented:
> In my exam summary, the bonus points are not shown. (Exercise 1 should have bonus points)
> ![image](https://user-images.githubusercontent.com/44401560/97865755-fb6c7680-1d0a-11eb-823a-83a7d59200e9.png)
> Mini remark: You could switch the white space to the second string, so without the second string, the square brackets are even:
> ![image](https://user-images.githubusercontent.com/44401560/97866103-8c435200-1d0b-11eb-8633-5ecc6034cb0a.png)
> The rest looked very good to me!

@FuchsDominik I added this now:
<img width="1414" alt="image" src="https://user-images.githubusercontent.com/41108487/97883584-58752600-1d25-11eb-9b52-699353b3c7fa.png">

The issue with the whitespace comes from the formatting. I tried out to use prettier-ignore to fix the issue with the styling mentioned by @FrankeLukas. 

@MaisiKoleni I added the bonus points also in the exercise view :

<img width="1127" alt="image" src="https://user-images.githubusercontent.com/41108487/97883776-95411d00-1d25-11eb-91ef-b724db0241af.png">

Would be nice if you could recheck :)
